libraries/SRV_Channel/SRV_Channel_aux.cpp

```diff
@@ -595,3 +596,117 @@ void SRV_Channels::constrain_pwm(SRV_Channel::Aux_servo_function_t function)
         }
     }
 }
+
+/*
+  upgrade RC* parameters into SERVO* parameters. This does the following:
+
+  - copies MIN/MAX/TRIM values from old RC parameters into new RC* parameters and SERVO* parameters. 
+  - copies RCn_FUNCTION to SERVOn_FUNCTION
+  - maps old RCn_REV to SERVOn_REVERSE and RCn_REVERSE
+
+  aux_channel_mask is a bitmask of which channels were RC_Channel_aux channels
+
+  Note that this code is highly dependent on the parameter indexing of
+  the old RC_Channel and RC_Channel_aux objects.
+
+  If rcmap is passed in then the vehicle code also wants functions for
+  the first 4 output channels to be remapped
+
+  We return true if an upgrade has been done. This allows the caller
+  to make any vehicle specific upgrades that may be needed
+*/
+bool SRV_Channels::upgrade_parameters(const uint8_t rc_keys[14], uint16_t aux_channel_mask, RCMapper *rcmap)
+{
+    // use SERVO16_FUNCTION as a marker to say that we have run the upgrade already
+    if (channels[15].function.configured_in_storage()) {
+        // upgrade already done
+        return false;
+    }
+    
+    // old system had 14 RC channels
+    for (uint8_t i=0; i<14; i++) {
+        uint8_t k = rc_keys[i];
+        if (k == 0) {
+            // missing parameter. Some vehicle types didn't have all parameters
+            continue;
+        }
+        SRV_Channel &srv_chan = channels[i];
+        RC_Channel &rc_chan = RC_Channels::channels[i];
+        const struct mapping {
+            uint8_t old_index;
+            AP_Param *new_srv_param;
+            AP_Param *new_rc_param;
+            enum ap_var_type type;
+            bool is_reverse;
+        } mapping[] = {
+            { 0, &srv_chan.servo_min,  &rc_chan.radio_min,  AP_PARAM_INT16, false },
+            { 1, &srv_chan.servo_trim, &rc_chan.radio_trim, AP_PARAM_INT16, false },
+            { 2, &srv_chan.servo_max,  &rc_chan.radio_max,  AP_PARAM_INT16, false },
+            { 3, &srv_chan.reversed,   &rc_chan.reversed,   AP_PARAM_INT8,  true },
+            { 1, &srv_chan.function,   nullptr,             AP_PARAM_INT8,  false },
```

Comment:
its in fact correct. The reason is that function is index 1 in RC_Channel_aux, and trim is index 1 in RC_Channel. The subtle difference is handled by the if statement on the rc_chan parameter